collaboration-concept.md

```diff
@@ -0,0 +1,137 @@
+# Collaboration Concept
+
+## Microsoft VSCode Live Share Extension
+
+The official Microsoft [Live Share Extension](https://visualstudio.microsoft.com/services/live-share/) implements collaboration support for the following features:
+
+- Editor Syncing (Text + Client markers for each user)
+- Host Syncing (i.e. other users follow the host into their currently selected file + position)
+- Terminal Syncing (if enabled). Both input+output is synced
+- Debug Syncing (if enabled). Both users and host can step through the code as if it were on one machine. Breakpoints are synced as well.
+- Plugin Syncing (i.e. external user has access to all installed plugins/extensions of the host user)
+- Chat Functionality
+- Mechanism for making the API available to other extensions, see [vsls](https://www.npmjs.com/package/vsls)
+
+Similar to the tunnel feature, the connection between host and other users is accomplished using hole-punching.
+I.e. a single central server serves as an intermediary to exchange messages. No messages are being exchanged directly between users. 
```

Comment:
According to the docs they do send messages directly between host and users if the network allows it, see https://learn.microsoft.com/en-us/visualstudio/liveshare/reference/connectivity#connection-modes. However every connection is authenticated via the cloud, so it's never fully private.